homeassistant/components/wallbox/config_flow.py

```diff
@@ -47,14 +59,27 @@ async def async_step_user(self, user_input=None):
         errors = {}
 
         try:
-            info = await validate_input(self.hass, user_input)
+            await self.async_set_unique_id(user_input["station"])
+            if not self._reauth_entry:
+                self._abort_if_unique_id_configured()
+                info = await validate_input(self.hass, user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
+            if user_input["station"] == self._reauth_entry.data[CONF_STATION]:
+                self.hass.config_entries.async_update_entry(
```

Comment:
We  should validate the new credentials in the reauth flow before updating the config entry data.